### ISSUE TITLE
Accuracy of VisDA2017 only 70.76

### ISSUE THREAD

**Issue by miss-rain** [uta-smile/TVT#1]
**I have follow your code,  in VisDA2017, the accuracy only 70.76**

**Comment by viyjy** [comment]
Can you reproduce the results on other datasets?

**Comment by viyjy** [comment]
> **I have follow your code, in VisDA2017, the accuracy only 70.76**

BTW, can you show me the command you used?

**Comment by miss-rain** [comment]
python3 main.py --train_batch_size 32 --dataset visda17 --name visda --source_list /home/cy/data/visda/train/train_list.txt --target_list /home/cy/data/visda/validation/validation_list.txt --test_list /home/cy/data/visda/validation/validation_list.txt --num_classes 12 --model_type ViT-B_16 --pretrained_dir checkpoint/ViT-B_16.npz --num_steps 20000 --img_size 256 --beta 1.0 --gamma 0.01 --use_im

**Comment by miss-rain** [comment]
> Can you reproduce the results on other datasets?

I have only run in VisDA

**Comment by viyjy** [comment]
Thanks, let me check this.

**Comment by miss-rain** [comment]
> Thanks, let me check this.

Is ok?

**Comment by viyjy** [comment]
> > Thanks, let me check this.
> 
> Is ok?

I am not in campus now, so I tried to use VPN to access my machine. However, the link to download VPN is broken, so I submitted a ticket to the IT service, hope that they can fix this issue tomorrow.

**Comment by miss-rain** [comment]
when use **'Transferability Adaptation Module'**,  

why only **'k'**  need **'Domain Adversarial Loss'**?

**Comment by viyjy** [comment]
I don't understand what does 'k' means, can you clarify it?

**Comment by miss-rain** [comment]
TVT framework(**image.png** in this project) in the paper,

the **'TAM'** module (i.e., **'Transferable MSA'**),

img-->patch token--> generous 'q', 'k', 'v',

why only 'k' use 'Domain Adversarial Discriminator'?  i am not  understand.

**Comment by viyjy** [comment]
'q' is query of the [CLS] token, while 'k' is key of local patches. We use 'Domain Adversarial Discriminator' to identify transferable local patches.

BTW, I haven't received the response from IT service, maybe they are on their holidays.

**Comment by miss-rain** [comment]
> 'q' is query of the [CLS] token, while 'k' is key of local patches. We use 'Domain Adversarial Discriminator' to identify transferable local patches.
> 
> BTW, I haven't received the response from IT service, maybe they are on their holidays.

**Is 'v' equal 'k' ? ( i.e., 'v' and  'k' both of local patches, and have same effect)**

**Comment by viyjy** [comment]
no, 'v' is value of local patches.

**Comment by miss-rain** [comment]
'v'(value) and 'k'(key) , they are all multiplied by a matrix,

why only 'k' use 'Domain Adversarial Discriminator', while 'v' not?

**Comment by viyjy** [comment]
because we usually measure the 'similarity' between 'q' and 'k' in attention mechanism, then the similarity is used to aggregate 'v'. You can check equation 1 in [Attention is all you need](https://arxiv.org/pdf/1706.03762.pdf)

**Comment by miss-rain** [comment]
> 

Fine.

**Comment by miss-rain** [comment]
python3 main.py --train_batch_size 32 --dataset visda17 --name visda --source_list /home/cy/data/visda/train/train_list.txt --target_list /home/cy/data/visda/validation/validation_list.txt --test_list /home/cy/data/visda/validation/validation_list.txt --num_classes 12 --model_type ViT-B_16 --pretrained_dir checkpoint/ViT-B_16.npz --num_steps 20000 --img_size 256 --beta 1.0 --gamma 0.01 --use_im


**This command mean train source model and adaption target model simultaneously?**

**(I do not find source only accuracy of 73.22% in visda2017)**

**Comment by miss-rain** [comment]
Domain adaption setting, should train a source model, and use this trained source model to init target model, finally, train target model for adaption.

**Comment by viyjy** [comment]
> 

It trains a model by using both source data and target data. It is not source-only

**Comment by viyjy** [comment]
> 

Not necessary. We can directly train a model which leverages both source data and target data to learn domain-invariant features

**Comment by miss-rain** [comment]
In my experiment, I use pre-trained  model (ViT-B_16.npz), 

in visda 2017

the accuracy of source only just 65%, not get 73.22% in your paper

**Comment by viyjy** [comment]
Sorry for the late reply. Did you try the pre-trained model on the source data?

**Comment by miss-rain** [comment]
I use pre-trained Vit-16-B, and get 71.3%  (source only)

more important ,  I still get **'Accuracy of VisDA2017 only 70.76'**.

**Comment by viyjy** [comment]
I rerun the code and can reproduce the result. Can you provide the screenshot of the printing output?

**Comment by miss-rain** [comment]
![Uploading TVT.png…]()

**Comment by miss-rain** [comment]
<img width="1266" alt="TVT" src="https://user-images.githubusercontent.com/17979331/148496298-caaeeee3-7349-40e2-a394-09c701d46c71.png">

**Comment by viyjy** [comment]
This is what I get:
![image](https://user-images.githubusercontent.com/20442927/148587692-e11d2aaf-4359-4cb7-b424-a3e7a8be2938.png)

**Comment by viyjy** [comment]
It seems like the validation loss in your experiment is quite large
